### DISCUSSION TITLE
Weird error description when pre-rendering

### DISCUSSION THREAD

**Discussion by alexbjorlig** [sveltejs/kit#8090] [Q&A]
Hi there.

I'm running the latest version of Sveltekit (v582). I can build my app locally on my machine, but when I push to Github actions I get this error message on build:

```
Error: ENOENT: no such file or directory, open '/home/runner/work/eddystone/eddystone/apps/web-app/.svelte-kit/generated/prerendered.json'
web-app:build:     at Object.openSync (node:fs:590:3)
web-app:build:     at Object.readFileSync (node:fs:458:35)
web-app:build:     at ChildProcess.<anonymous> (file:///home/runner/work/eddystone/eddystone/node_modules/@sveltejs/kit/src/exports/vite/index.js:478:38)
web-app:build:     at ChildProcess.emit (node:events:513:28)
web-app:build:     at Process.ChildProcess._handle.onexit (node:internal/child_process:293:12) {
web-app:build:   errno: -2,
web-app:build:   syscall: 'open',
web-app:build:   code: 'ENOENT',
web-app:build:   path: '/home/runner/work/eddystone/eddystone/apps/web-app/.svelte-kit/generated/prerendered.json'
web-app:build: }
```

Any hints for what I could try - what is `prerendered.json`? 

**solved**
Turns out that my server could not start for prerendering in CI - a better error message would be nice - but it works now.

**Comment by danielzting** [comment]
I'm having the same issue in GitLab, how did you end up fixing this?

**Reply by danielzting** [reply]
Turned out SvelteKit was somehow trying to run my database connection code when prerendering (I use Prisma). Despite setting `prerendered = false` in the root layout and `entries = []` in `svelte.config.js`, the prerendering process was still trying to run for some reason.

I finally got around this by not connecting if the code is being run in the context of a build. (Still not sure why building my app actually runs my code.)
```ts
import { PrismaClient } from '@prisma/client';
import { building } from '$app/environment';

const prisma = new PrismaClient();
if (!building) {
	await prisma.$connect();
}
export default prisma;
```

**Reply by geoffrich** [reply]
See the [docs](https://kit.svelte.dev/docs/building-your-app#during-the-build) - SvelteKit needs to load your routing files at buildtime to determine which [page options](https://kit.svelte.dev/docs/page-options) are set

**Reply by danielzting** [reply]
You know what they say, "a few hours of trial and error can save you several minutes of looking at the README"

Thanks 🤡

**Reply by geoffrich** [reply]
To be fair, this doc was added yesterday!
